src/crystal/system/win32/file.cr

```diff
@@ -26,11 +26,84 @@ module Crystal::System::File
   end
 
   def self.open(filename : String, flags : Int32, perm : ::File::Permissions) : {LibC::Int, Errno}
-    flags |= LibC::O_BINARY | LibC::O_NOINHERIT
```

Comment:
We should keep these

Reply:
Ensuring `O_BINARY` is added should address your below comment.

`O_NOINHERIT` seems to be controlled via the `bInheritHandle` property of https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/aa379560(v=vs.85) which is the 4th argument to `CreateFileW`. However, the documentation also says:

> If this parameter is NULL, the handle returned by CreateFile cannot be inherited by any child processes the application may create and the file or device associated with the returned handle gets a default security descriptor.

So given we'd always be applying `O_NOINHERIT`, I think we'll be fine as it is.